valhalla/midgard/encoded.h

```diff
@@ -1,22 +1,110 @@
-#ifndef VALHALLA_MIDGARD_ENCODED_H_
-#define VALHALLA_MIDGARD_ENCODED_H_
-
-#include <valhalla/midgard/shape_decoder.h>
+#pragma once
 
 #include <cmath>
+#include <stdexcept>
 #include <string>
 #include <type_traits>
 #include <vector>
 
+// we store 6 digits of precision in the tiles, changing to 7 digits is a breaking change
+// if you want to try out 7 digits of precision you can uncomment this definition
+//#define USE_7DIGITS_DEFAULT
+#ifdef USE_7DIGITS_DEFAULT
+constexpr double DECODE_PRECISION = 1e-7;
+constexpr int ENCODE_PRECISION = 1e7;
+constexpr size_t DIGITS_PRECISION = 7;
+#else
+constexpr double DECODE_PRECISION = 1e-6;
+constexpr int ENCODE_PRECISION = 1e6;
+constexpr size_t DIGITS_PRECISION = 6;
+#endif
+
 namespace valhalla {
 namespace midgard {
 
+template <typename Point> class Shape7Decoder {
```

Comment:
i ditched the other header that had this info and just moved it to this one. its nice to have all the polyline encoding/decoding in one place